src/main/perl/ComponentProxy.pm

```diff
@@ -21,7 +22,7 @@ my $ec = LC::Exception::Context->new->will_store_all;
 
 use Readonly;
 Readonly my $COMPONENTS_PROFILE_PATH => '/software/components';
-Readonly my $RUN_FROM => '/tmp';
+Readonly my $RUN_FROM => File::Temp->newdir('/tmp/ncd-components-XXXXXXXX');
```

Comment:
can you add `CLEANUP => 1` (i'm aware it's the default, but it's better if it is made explicit)
@ttyS4 @ned21 you are aware that this removes EL5 support? or requires a recent `FILE::Temp` version?

Reply:
see also `CAF::Path` https://github.com/quattor/CAF/blob/master/src/main/perl/Path.pm#L383.
it support temp directory creation and error handling

Reply:
Sure, i can add `CLEANUP` to the call.
For RH5, I think it should just work fine. I can test tomorrow.

So you want me to rework to use CAF::Path (it uses File::Temp itself) or it is optional?

Reply:
no, i'm pointing out that in CAF::Path newdir isn't used because of EL5 support. correct ref is https://github.com/quattor/CAF/blob/master/src/main/perl/Path.pm#L409
but feel free to use CAF::Path. that way we can add unittests more easily.